### PR TITLE
bugfix/20435-point-events-update

### LINKED ISSUE #20435: Point events update does not work.
#### Expected behaviour
Even if the event has been triggered before, using `series.update` with the appropriate options (as in the demo) should update the event.

#### Actual behaviour
Once a function passed to the `series.point.events.mouseOver` option (or other, e.g. `click`, etc.) is called, it cannot be changed or removed by updating it.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/5ywL7zqv/
1. Open console
2. Click the button to add/remove an event; hover over points to see if the event is still working.

#### Product version
Highcharts v11.3.0

### PR DESCRIPTION
**Pull request by hubertkozik**
Fixed #20435, point events update didn't work.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206339642151373

### CODE REVIEW

File: ts/Core/Series/Point.ts [hunk 1]
```diff
@@ -1301,33 +1292,45 @@ class Point {
...
-                    point.options
-                ),
-                events = options.events;
-
-            point.events = events;
+    public manageEvent(eventType: string): void {
+        const point = this,
+            options = merge(
+                point.series.options.point as PointOptions,
+                point.options
```
Comment (bre1470) [inline ts/Core/Series/Point.ts:1305]: Shouldn't `point.options` already include `SeriesOptions.point` when instantiated? Or is this about `Series.update` follow-ups?

Comment (hubertkozik) [reply ts/Core/Series/Point.ts]: Only `point.options` doesn't contain every events, but `point.series.options.point` indeed contains everything. I changed that part of the code a little bit.

**Review by bre1470 — APPROVED** [review]
LGTM 👍

**Review by goransle — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks, looks good 👍

**Pull request merged**